### ISSUE TITLE
Allow for whitespace in `* except (...)`

### ISSUE THREAD

**Issue by nfcampos** [tconbeer/sqlfmt#219]
**Describe the bug**
See the 2 repros below, which I hoped would result in the same output, but they don't. Why is this important? Because when eg converting an existing codebase/project to use `sqlfmt` I have 1. no control over what the existing format is, 2. no desire to manually change it (expect `sqlfmt` to do that for me)

**To Reproduce** A
If I start with this output and format it
```
select
    recentagg_root.*
    except (recentagg),
    (
        select as struct
            recentagg.*,
            recentagg_topphases_array.topphases,
            recentagg_topsymptoms_array.topsymptoms,
            recentagg_toppositivesymptoms_array.toppositivesymptoms,
            recentagg_topactivities_array.topactivities,
    ) recentagg,
from recentagg_root
left join recentagg_topphases_array using(id)
left join recentagg_topsymptoms_array using(id)
left join recentagg_toppositivesymptoms_array using(id)
left join recentagg_topactivities_array using(id)
```

**Actual behavior** A
Then I get
```
select recentagg_root.*
except
(recentagg),
(
    select as struct
        recentagg.*,
        recentagg_topphases_array.topphases,
        recentagg_topsymptoms_array.topsymptoms,
        recentagg_toppositivesymptoms_array.toppositivesymptoms,
        recentagg_topactivities_array.topactivities,
) recentagg,
from recentagg_root
left join recentagg_topphases_array using(id)
left join recentagg_topsymptoms_array using(id)
left join recentagg_toppositivesymptoms_array using(id)
left join recentagg_topactivities_array using(id)
```

**To Reproduce** B
If instead I start with this (ie. `except` in the same line as the `*`)
```
select
    recentagg_root.* except (recentagg),
    (
        select as struct
            recentagg.*,
            recentagg_topphases_array.topphases,
            recentagg_topsymptoms_array.topsymptoms,
            recentagg_toppositivesymptoms_array.toppositivesymptoms,
            recentagg_topactivities_array.topactivities,
    ) recentagg,
from recentagg_root
left join recentagg_topphases_array using(id)
left join recentagg_topsymptoms_array using(id)
left join recentagg_toppositivesymptoms_array using(id)
left join recentagg_topactivities_array using(id)
```

**Actual behavior** B
Then I get what I actually want (which happens to be the same as the starting point, but that's just me trying to make the repro minimal)
```
select
    recentagg_root.* except (recentagg),
    (
        select as struct
            recentagg.*,
            recentagg_topphases_array.topphases,
            recentagg_topsymptoms_array.topsymptoms,
            recentagg_toppositivesymptoms_array.toppositivesymptoms,
            recentagg_topactivities_array.topactivities,
    ) recentagg,
from recentagg_root
left join recentagg_topphases_array using(id)
left join recentagg_topsymptoms_array using(id)
left join recentagg_toppositivesymptoms_array using(id)
left join recentagg_topactivities_array using(id)
```

**Additional context**
What is the output of `sqlfmt --version`?
Commit `55d3b361843c5022890265aa725fcd2719add614`

**Comment by tconbeer** [comment]
Thanks for the report, @nfcampos!

I think this is a bug in our lexing/parsing logic. When we lex the token `except`, we need to determine whether it follows a `*`, or whether it's the set operator (`except` is an overloaded keyword and can also be used in place of `union`). I think we're not accounting properly for newlines/whitespace in that logic. If that's the cause, this should be an easy fix. I'll target 0.10.0 or 0.10.1 for this.

And yes, philosophically I agree with you -- the formatting of the input file shouldn't matter very much at all!

**Comment by tconbeer** [comment]
@nfcampos This is fixed and included in the 0.10.0 release, which should be hitting PyPI momentarily. Thanks for using sqlfmt!

**Comment by nfcampos** [comment]
@tconbeer amazing, thank you
